tools/system_libs.py

```diff
@@ -1340,6 +1340,9 @@ class libmalloc(MTLibrary):
   name = 'libmalloc'
 
   cflags = ['-fno-builtin']
+  # malloc/free/calloc are runtime functions and can be generated during LTO
```

Comment:
Interesting... how can LLVM generate a new malloc call during LTO? Does it sometimes move a stack allocation to the heap..? (but that seems bad)

Reply:
I'm not exactly sure.. but this fixes he repro case that was provided.

Looking at the llvm source code it looks like there is maybe just one pass that generates calloc calls:

```
$ git grep emitCalloc
llvm/include/llvm/Transforms/Utils/BuildLibCalls.h:  Value *emitCalloc(Value *Num, Value *Size, IRBuilderBase &B,
llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp:      emitCalloc(ConstantInt::get(IRB.getIntPtrTy(DL), 1),
llvm/lib/Transforms/Utils/BuildLibCalls.cpp:Value *llvm::emitCalloc(Value *Num, Value *Size, IRBuilderBase &B,
```

Reply:
```
  /// If we have a zero initializing memset following a call to malloc,          
  /// try folding it into a call to calloc.                                      
  bool tryFoldIntoCalloc(MemoryDef *Def, const Value *DefUO) {  
```

Reply:
Interesting, thanks... so it's an existing malloc turning into a calloc.

Reply:
@kripken clang will turn malloc+memset pairs into calloc with -O1: https://www.godbolt.org/z/8r3Ydjs6a (I was scratching my head for quite a while on that topic, because the code which triggers the error didn't have a single calloc in it)

Reply:
@floooh Heh, yeah, that is quite a confusing optimization if you don't expect it... and I wouldn't have :smile: